src/js/components/accordion.js

```diff
@@ -0,0 +1,270 @@
+    this._handleKeydown = this._handleKeydown.bind(this);
+
+    this.init();
+  }
+
+  /**
+   * This function is used to handle all click events on
+   * the document. It accepts the Event object, checks target to see if it came
+   * from an accordion trigger. From there, it gets the related panel and opens
+   * it.
+   * @param {Event} event
+   */
+  _handleClick(event) {
+    const currentTrigger = event.target.closest(this.triggerSelector);
+    // If not an accordion trigger, ignore
+    if (!currentTrigger) return;
+
+    // Get the data-accordion-trigger value
+    const currentTriggerValue = currentTrigger.getAttribute(this.triggerAttribute);
+
+    // Get the associated panel
+    const currentPanel = this.element.querySelector(`[${this.panelAttribute}="${currentTriggerValue}"]`);
+
+    // Open or close accordion
+    currentTrigger.getAttribute('aria-expanded') === 'true' ? this.close(currentPanel) : this.open(currentPanel);
+
+  }
+
+  /**
+   * This function is used to handle all keydown events on the document. It
+   * sets up handling for up and down arrows.
+   * @param {Event} event
+   */
+  _handleKeydown(event) {
+    if (event.keyCode === keyCodes.up || event.keyCode === keyCodes.down || event.keyCode === keyCodes.end || event.keyCode === keyCodes.home) {
+      const accordionParent = event.target.closest('[data-accordion]');
+      // If not an accordion, ignore
+      if (!accordionParent) return;
+
+      const currentTrigger = event.target.closest(this.triggerSelector);
+      if (!currentTrigger) return;
+
+      // Create an array of all the focusable elements within the accordion
+      const nextTrigger = this.triggers.indexOf(currentTrigger) + 1;
+      const prevTrigger = this.triggers.indexOf(currentTrigger) - 1;
+
+      switch (event.keyCode) {
+        case keyCodes.up: {
+          if (this.triggers[prevTrigger] === undefined) {
+            this.triggers[this.triggers.length - 1].focus();
+          } else {
+            this.triggers[prevTrigger].focus();
+          }
+          break;
+        }
+
+        case keyCodes.down: {
+          if (this.triggers[nextTrigger] === undefined) {
+            this.triggers[0].focus();
+          } else {
+            this.triggers[nextTrigger].focus();
+          }
+          break;
+        }
+
+        case keyCodes.end: {
+          this.triggers[this.triggers.length - 1].focus();
+          break;
+        }
+
+        case keyCodes.home: {
+          this.triggers[0].focus();
+          break;
+        }
+
+        default: {
+          break;
+        }
+      }
+    }
+  }
+
+  /**
+   * This function is used to initialize accordions on load. It also adds the
+   * appropriate aria-expanded, and hidden attributes to remove some burden
+   * from the develop.
+   */
+  _openOnInit() {
+    // If accordion is set to open all panels on init, open them
+    if (this.element.hasAttribute('data-accordion-init')) {
```

Comment:
In the side nav we are offering an option `openAllOnItinit` options as part of the second argument to the component construction (options object). I wonder if it make sense to do that here instead of using a data attribute? Either way the API for how we off this kind of "open all on init" functionality should be consistent. either attributes for properties on the config object (second argument to the constructor).